slick/src/main/scala/slick/jdbc/SQLiteProfile.scala

```diff
@@ -97,9 +97,10 @@ trait SQLiteProfile extends JdbcProfile {
   class ModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext) extends JdbcModelBuilder(mTables, ignoreInvalidDefaults) {
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
       /** Regex matcher to extract name and length out of a db type name with length ascription */
-      final val TypePattern = "^([A-Z]+)(\\(([0-9]+)\\))?$".r
+      final val TypePattern = """^([A-Z]+)(|\(([0-9]+)(|,([0-9]+))\))?$""".r
```

Comment:
I don't think `DECIMAL(1,)` should be supported, so only using `?` as optional, this would be correct: `^([A-Z]+)(\(([0-9]+)(,([0-9]+))?\))?$` it does not allow coma without second number, and will add both 2nd and 4th capture groups as null if missing, only meaningful capture groups are 1 (DECIMAL), 3 (first number), 5 (second number), if numbers are missing they are null